### DISCUSSION TITLE
这个代码演示插件的 json 配置支持哪些呀，有可供参考的文档吗？

### DISCUSSION THREAD

**Discussion by Fuphoenixes** [vuepress-theme-hope/vuepress-theme-hope#1269] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/27912232/145983426-af85c983-7b86-42bb-b8f9-739634904563.png)

**Comment by Mister-Hope** [comment, marked as answer]
支持几乎全部的 codeDemoOptions，详见 https://vuepress-theme-hope.github.io/md-enhance/zh/config/#demo

**Comment by Fuphoenixes** [comment]
还是有些疑问，我其实是想用这个插件展示我自定义的组件，我需要怎么引入这个组件呢？
![image](https://user-images.githubusercontent.com/27912232/148043547-9a0542b3-82c0-4353-b6c0-30ea1ed4b3a6.png)

**Reply by Mister-Hope** [reply]
你正常在markdown引入就好了。

这个功能是让你写一些代码，直接跑在用户的浏览器上进行演示。而不是在markdown中调用并显示注册在vuepress实例上的组件，并展示调用代码。

1、2月份我有计划考虑如何实现类似的功能，但可能会以别的方式。

主要的问题原因是 vuepress 的确可以解析 markdown 中的 script 块，但插件只能拿到文件内容，也就是你文件的字符串版本，换言之插件需要自己实现你代码的渲染和高凉，同时要保证vuepress在对markdown结果当作页面进行后续渲染的时候，你的组件能够正确的渲染并调用。这在实现上更为复杂，而且它是没有办法拿到一个全局注册组件（已经解析为 javascript 对象）的源代码的。
